services/QuillLMS/config/initializers/rails_admin.rb

```diff
@@ -1,146 +1,432 @@
+      eager_load :concepts
+    end
+    include_all_fields
+  end
+
+  config.model 'SkillGroup' do
+    list do
+      exclude_fields :skill_group_activities, :activities, :skills
+    end
+  end
+
+  config.model 'Standard' do
+    list do
+      exclude_fields :activities, :change_logs
+    end
+  end
+
+  config.model 'StandardCategory' do
+    list do
+      exclude_fields :activities, :change_logs
+      field :standards do
+        eager_load :standards
       end
     end
+  end
 
-    config.model Activity do
-      edit do
-        exclude_fields :classroom_units, :classrooms, :unit_activities, :units
+  config.model 'StandardLevel' do
+    list do
+      exclude_fields :activities, :change_logs
+      field :standards do
+        eager_load :standards
       end
-      list do
-        field :id
-        field :name do
-          searchable true
-        end
-        field :classification
-        field :flags
+    end
+  end
+
+  config.model 'Topic' do
+    list do
+      exclude_fields :activities, :activity_topics, :change_logs
+    end
+  end
+
+  config.model 'TeacherInfo' do
+    list do
+      exclude_fields :teacher_info_subject_areas, :subject_areas
+    end
+  end
+
+  config.model 'Unit' do
+    list do
+      exclude_fields :activities, :standards
+      configure :classrooms do
+        eager_load true
       end
+      limited_pagination true
     end
+    edit do
+      exclude_fields :user
+    end
+  end
 
-    # Limit pagination for models with large datasets (~1M+) because of performance
-    config.model ClassroomActivity do
-      list { limited_pagination true }
+  config.model 'UnitTemplate' do
+    list do
+      exclude_fields :activities, :activities_unit_templates, :partner_content, :recommendations
     end
+  end
 
-    config.model UnitActivity do
-      list { limited_pagination true }
+  config.model 'UnitTemplateCategory' do
+    list do
+      exclude_fields :unit_templates
     end
+  end
 
-    config.model ClassroomUnit do
-      list { limited_pagination true }
+  config.model 'User' do
+    list do
+      field :name do
+        searchable true
+      end
+      field :email do
+        searchable true
+      end
+      field :password_digest do
+        searchable false
+      end
+      field :created_at
+      field :account_type
+      field :role
+      limited_pagination true
     end
 
-    # Exclude the models with huge datasets (~10M+) because they are performance hits
-    MODELS_TO_EXCLUDE = [
-      'ActivitySession',
-      'OldConceptResult',
-      'Notification',
-      'OauthAccessToken',
-      'OldActivitySession',
-      'StudentClassroom'
-    ]
+    edit do
+      field :name
+      field :email
+      field :username
+    end
+  end
 
-    config.excluded_models.push(*MODELS_TO_EXCLUDE)
+  config.model 'UserSubscription' do
+    field :user
+    field :subscription do
+      searchable [:id, :account_type]
+    end
   end
+
+  config.included_models = [
```

Comment:
This is an exported list of `ActiveRecord `descendants. I decided to use an `allow list` instead of a `block list`, since I think it's better to be explicit about the models we want to show, so if we add a new large model, it doesn't cause performance issues that surprise us. The downside is we'd have to add models here as they are added. Open to feedback.

Also, while editing this a few times, I found having the full list of everything with the unused commented out we easier for me to parse, but others may not like this. Open to feedback.

Reply:
I like the white-list approach over the black-list approach here.  Agree that maintenance may be an ongoing issue, but I half-suspect that even as we add more models, it will turn out that very few of them will need to be accessible through RailsAdmin.  So even if we never add another model to this list, it might turn out not to matter.

I'm okay with the commenting-out thing, but think it may be worth having an explanatory comment at the top of the list so that we don't come back in six months and just delete them.  The fact that it serves as a record that "we made a decision about this model" will also probably make it easier if we do need to come back later and add new models to the list.  The distinction between "this model hasn't been added because we haven't looked at it yet" and "this model is intentionally excluded" seems valuable to me.